src/git.ts

```diff
@@ -0,0 +1,53 @@
+import * as github from '@actions/github'
+import * as core from '@actions/core'
+import { minimatch } from 'minimatch'
+
+const FILE_PATTERNS = [
+  '**/*.js',
+  '**/*.jsx',
+  '**/*.tsx',
+  '**/*.esm',
+  '**/*.html',
+  '**/*.htm',
+  '**/*.vue',
+  '**/*.md',
+  '**/*.markdown'
+] as const
+
+export async function getChangedFiles(token: string): Promise<string[]> {
+  const octokit = github.getOctokit(token)
+  const { context } = github
+
+  if (!context.payload.pull_request) {
+    core.debug('Not a pull request, checking push diff')
+    const base = context.payload.before
+    const head = context.payload.after
+
+    const response = await octokit.rest.repos.compareCommits({
+      owner: context.repo.owner,
+      repo: context.repo.repo,
+      base,
+      head
+    })
+
+    return (
+      response.data.files
+        ?.filter((file) =>
+          FILE_PATTERNS.some((pattern) => minimatch(file.filename, pattern))
```

Comment:
Does `minimatch` do case insensitive search? Will `readme.MarKdoWN` be matched here?